server/src/main/java/io/crate/expression/tablefunctions/SingleColumnRangeIterator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.tablefunctions;
+
+
+import io.crate.data.Row;
+import io.crate.data.RowN;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+
+
+public final class SingleColumnRangeIterator<T, U extends Number> implements Iterator<Row> {
```

Comment:
I think the `U extends Number` constraint is unecessary, no?

Also not sure about the name. I think it emphases the `SingleColumn` part too much. How about just `RangeIterator`, `SeriesGenerator`, or `GenerateSeriesIterator` ?

Also, could this an `Iterable`? Would make it a bit more powerful